### ISSUE TITLE
Thumbnail corruption and subsequent crashing

### ISSUE THREAD

**Issue by FrostKiwi** [TeamNewPipe/NewPipe#4516]
I switched to Android 10 on a f2fs system and cache partition, up from Android 7 yesturday.
NewPipe suddenly behaves as if it were inflicted with memory corruption.
If I look at a list of Videos everything looks fine. As soon as I load a video and close it again to return to the Main page of the video, the thumbnail is corrupted with the tell tale streaks of overwritten memory.
I checked both 0.19.8 and 0.20.0, same deal.

Now it get's to where it becomes a problem: The subs / "What's new" page. The thumnails are saved by NewPipe to prevent redownloading. But the thumbnail stays corrupted. It is apparently saved in a corrupted state. If I load up the subs page, the app crashes instantly. No Error, no message, no logs. Just straight closed and that's it.
Any other aspect of the app works fine. But the corrputed thumbnail in the subs page prevents me from opening that page.

### Steps to reproduce the bug
NewPipe is the only App showing problems :S
Open up any Video, close it, watch the thumbnail get corrupted...

### Logs
Unfortunately NewPipe does not offer the logs dialog after the crash happened.

### Device info
 - Android version/Custom ROM version: (LineageOS 17.1, Argon kernel)
 - [Link to Rom for context](https://forum.xda-developers.com/oneplus-one/development/rom-unofficial-optimized-lineageos-14-1-t3693223)
 - Device model: OnePlus One (codnamed bacon or A0001)

### Screenshots/Screen recordings
![Screenshot_20201015-113042_NewPipe](https://user-images.githubusercontent.com/60887273/96107173-63822680-0edc-11eb-9baa-2c75b1ac070e.png)
![Screenshot_20201015-113627_NewPipe](https://user-images.githubusercontent.com/60887273/96107185-667d1700-0edc-11eb-9cd1-fed3ae6b6bfb.png)
![Screenshot_20201015-113651_NewPipe](https://user-images.githubusercontent.com/60887273/96107193-6846da80-0edc-11eb-8a6f-cc11ef11aaab.png)
![Screenshot_20201015-113701_NewPipe](https://user-images.githubusercontent.com/60887273/96107199-6aa93480-0edc-11eb-80c0-ac55fa832b56.png)

**Comment by opusforlife2** [comment]
Could you try "Wipe cached metadata" from the history settings?

**Comment by FrostKiwi** [comment]
> 
> 
> Could you try "Wipe cached metadata" from the history settings?

Thx for the suggestion.
Tried it. It did not resolve the crashing from the "What's New" page. The Thumbnail of the watched video remains blank for a fraction of a second before the crash occurs.
Unless the big switch to 0.20.0 messed up what "wipe cached metadata" does, this did not resolve the crashing.

**Comment by wb9688** [comment]
Could you also wipe cache in Android settings for NewPipe? Btw do you have any other device? If so, could you export database in NewPipe's settings and import it on another Android device?

**Comment by FrostKiwi** [comment]
> ~~Unless the big switch to 0.20.0 messed up what "wipe cached metadata" does~~ , this did not resolve the crashing.

Nope, same deal on 0.19.8

> 
> 
> Could you also wipe cache in Android settings for NewPipe? Btw do you have any other device? If so, could you export database in NewPipe's settings and import it on another Android device?

Since everyone else is out of house, no luck with another device. I did the following: I created a NewPipe database with just one Subscribtion, corrupted a thumbnail and exported.
Importing this database into the Bluestacks emulator did not change anything, though Bluestacks is very divorced from being an android device. More interestingly, I deleted storage and chache from new pipe and reimported the database into the same app. The subs screen remains broken in this case!
I also did the reverse. Created the same database with the same subscripton, but did not corrupt a thumbnail. Deleted storage, important uncorrupted database and it works!
Lastly, overwriting works as well. Imported corrupted database, crash, then important clean database without clearing storage and it works again! Facinating. So whatever is corrupted, stays in the database file, as linked to this post.
[corrupted.zip](https://github.com/TeamNewPipe/NewPipe/files/5384282/corrupted.zip)
[clean.zip](https://github.com/TeamNewPipe/NewPipe/files/5384283/clean.zip)

**Comment by FrostKiwi** [comment]
For the sake of completness, here is what a screenrecording of the crash looks like. I used the database from corrupted.zip, where the thumbnail of the newest vido of the subscription to my channel is corrupted. Opening the "what's new" page shows the thumbnail as empty and the app crashes half a second later.
(video zipped, since github does not allow mp4)
[ScreenRecord-2020-10-15-13-04-23.zip](https://github.com/TeamNewPipe/NewPipe/files/5384315/ScreenRecord-2020-10-15-13-04-23.zip)

**Comment by Tomoms** [comment]
Hello, I'm the developer of the ROM @FrostKiwi is using. I too face the same issue (on the same device, and Android version, of course). However, I have noticed that on my device not all thumbnails look corrupted - only a few of them, and they're always the same ones (i.e. I tried with video A, B, C, D, E, and the only corrupted thumbnail was D's - the other ones always looked fine).

**Comment by wb9688** [comment]
@Tomoms: Does it also happen when using other apps that use [Universal Image Loader](https://github.com/nostra13/Android-Universal-Image-Loader)? For example F-Droid uses it as well.

**Comment by FrostKiwi** [comment]
> 
> 
> @Tomoms: Does it also happen when using other apps that use [Universal Image Loader](https://github.com/nostra13/Android-Universal-Image-Loader)? For example F-Droid uses it as well.

By F-Droid's use, I presume you are reffering to the screenshotsdisplayed on the App pages. Seems to be all fine. Some images are slow to load and don't display for a minute, but I chalk this up to F-Droid servers being slow.
![Screenshot_20201015-142742_F-Droid](https://user-images.githubusercontent.com/60887273/96123285-e6ae7700-0ef2-11eb-8136-3564b527c004.png)
![Screenshot_20201015-142847_F-Droid](https://user-images.githubusercontent.com/60887273/96123299-ec0bc180-0ef2-11eb-9188-9da16d08793c.png)
![Screenshot_20201015-142909_F-Droid](https://user-images.githubusercontent.com/60887273/96123305-ee6e1b80-0ef2-11eb-949a-ca3261441031.png)

**Comment by Tomoms** [comment]
> @Tomoms: Does it also happen when using other apps that use [Universal Image Loader](https://github.com/nostra13/Android-Universal-Image-Loader)? For example F-Droid uses it as well.

Yes, F-Droid works fine, all screenshots are displayed properly.

**Comment by wb9688** [comment]
Btw could you test if [this APK](https://github.com/TeamNewPipe/NewPipe/files/5384864/app-debug.apk.zip) works properly? It contains Universal Image Loader built from master instead of the latest release.

**Comment by FrostKiwi** [comment]
> 
> 
> Btw could you test if [this APK](https://github.com/TeamNewPipe/NewPipe/files/5384864/app-debug.apk.zip) works properly? It contains Universal Image Loader built from master instead of the latest release.

Yeeeep, same deal. Cleared everything. Thumbnails still corrupt and the "What's new" still crashes by saving the corrupted thumbnail.

**Comment by Tomoms** [comment]
> Btw could you test if [this APK](https://github.com/TeamNewPipe/NewPipe/files/5384864/app-debug.apk.zip) works properly? It contains Universal Image Loader built from master instead of the latest release.

Yes, still not working properly. I'm about to test it on another device, give me a couple of minutes.

**Comment by wb9688** [comment]
>Yeeeep, same deal. Cleared everything. Thumbnails still corrupt and the "What's new" still crashes by saving the corrupted thumbnail.

Hmm… I guess in that case we'll have to look at what F-Droid and NewPipe do differently.

@FrostKiwi: As for the crash, do you also not get a stack trace in logcat?

**Comment by Tomoms** [comment]
> > Yeeeep, same deal. Cleared everything. Thumbnails still corrupt and the "What's new" still crashes by saving the corrupted thumbnail.
> 
> Hmm… I guess in that case we'll have to look at what F-Droid and NewPipe do differently.
> 
> @FrostKiwi: As for the crash, do you also not get a stack trace in logcat?

Breaking news: I cannot reproduce the issue using the debug apk on another device. I guess there's something faulty in my ROM then (?)

**Comment by wb9688** [comment]
@Tomoms: Probably an issue in your ROM, as I've personally never experienced in NewPipe, however it's still weird that it doesn't happen in F-Droid. If you have time for it, could you please check what F-Droid does differently from NewPipe with UIL and find a way to fix it for NewPipe?

**Comment by FrostKiwi** [comment]
> @FrostKiwi: As for the crash, do you also not get a stack trace in logcat?

Will do this evening. Gotta go now unfortunately.

> Breaking news: I cannot reproduce the issue using the debug apk on another device. I guess there's something faulty in my ROM then (?)

I always compile stuff with -Ofast and get stirn looks from collegues for being such a dare devil^^
Maybe this is some unkown edge case caused by `all C/C++ components are build with -O3/-g0 flags`
Though this does indeed only happen in NewPipe, so it's not like it's completly the ROM's fault... just all a weird edge case we are triggering...

**Comment by FrostKiwi** [comment]
@Tomoms 
I may have been a bit too hasty with NewPipe's judgement. I have now seen the same corruption happening in another App. "[Ebay Kleinanzeigen](https://play.google.com/store/apps/details?id=com.ebay.kleinanzeigen)" (Germany's Craigslist), formerly known as kijiji, shows the same corruption when loading photos and crashes seconds after.
The App actually managed to save the corrupted photo. (White square is my censoring) I will still do logcat for completness sake later on. This photo looks like what normally happens when I poke around in i915 intel driver too much :D
![G_1602776461642_55668](https://user-images.githubusercontent.com/60887273/96155653-731b6280-0f10-11eb-861d-49752ca24c55.jpg)
![Screenshot_20201015-174137_eBay_Kleinanzeigen](https://user-images.githubusercontent.com/60887273/96155665-77e01680-0f10-11eb-86a1-256f1014ee78.png)

**Comment by FrostKiwi** [comment]
`Files and NewPipe crash for the same reason [...] I'll fix these issues in the next build` [Source](https://forum.xda-developers.com/showpost.php?p=83720793&postcount=1072)
So in the end it appears to be a ROM related issue. Though it is interesting, why it affected NewPipe, but not F-Droid.
For completness sake here is the logcat:
@11:14 I opened NewPipe, corrupted a thumbnail and caused it to crash.
Then I opened Ebay Kleinanzeigen and in a chat window submited an image, caused it to crash.
Lastly opened the file browser, which was in List view and it immediately crashed.
After this, I took the log.
[logcat2.txt](https://github.com/TeamNewPipe/NewPipe/files/5390580/logcat2.txt)
